params/upgrades_config.go

```diff
@@ -0,0 +1,252 @@
+// (c) 2022 Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package params
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ava-labs/subnet-evm/precompile"
+	"github.com/ava-labs/subnet-evm/utils"
+)
+
+// Upgrade is a helper struct embedded in UpgradesConfig, representing
+// each of the possible stateful precompile types that can be activated
+// through UpgradesConfig.
+type Upgrade struct {
+	ContractDeployerAllowListConfig *precompile.ContractDeployerAllowListConfig `json:"contractDeployerAllowListConfig,omitempty"` // Config for the contract deployer allow list precompile
+	ContractNativeMinterConfig      *precompile.ContractNativeMinterConfig      `json:"contractNativeMinterConfig,omitempty"`      // Config for the native minter precompile
+	TxAllowListConfig               *precompile.TxAllowListConfig               `json:"txAllowListConfig,omitempty"`               // Config for the tx allow list precompile
+	FeeManagerConfig                *precompile.FeeConfigManagerConfig          `json:"feeManagerConfig,omitempty"`                // Config for the fee manager precompile
+}
+
+type getterFn func(*Upgrade) precompile.StatefulPrecompileConfig
+
+// getters contains a list of functions that return each of the stateful precompile
+// types that can be retrieved from an upgrade.
+var getters = []getterFn{
+	func(u *Upgrade) precompile.StatefulPrecompileConfig { return u.ContractDeployerAllowListConfig },
+	func(u *Upgrade) precompile.StatefulPrecompileConfig { return u.ContractNativeMinterConfig },
+	func(u *Upgrade) precompile.StatefulPrecompileConfig { return u.TxAllowListConfig },
+	func(u *Upgrade) precompile.StatefulPrecompileConfig { return u.FeeManagerConfig },
+}
```

Comment:
Is the ordering of `getters` relevant?

Reply:
not really, switched to an enum / key based approach to avoid repeating the functions.